src/crimson/common/errorator.h

```diff
@@ -859,6 +860,44 @@ struct errorator {
     return make_ready_future<>();
   }
 
+  template <typename Iterator, typename Func>
+  static inline errorator<AllowedErrors...>::future<>
+  parallel_for_each(Iterator begin, Iterator end, Func&& func) noexcept {
+    parallel_for_each_state* s = nullptr;
+    // Process all elements, giving each future the following treatment:
+    //   - available, not failed: do nothing
+    //   - available, failed: collect exception in ex
+    //   - not available: collect in s (allocating it if needed)
+    while (begin != end) {
+      auto f = seastar::futurize_invoke(std::forward<Func>(func), *begin++);
```

Comment:
nit: just for the readability sake it might be helpful to advance the iterator in a dedicated line.